crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -347,3 +352,184 @@ async fn cow_amm(web3: Web3) {
+    services
+        .start_autopilot(
+            None,
+            vec![
+                "--drivers=test_solver|http://localhost:11088/test_solver,mock_solver|http://localhost:11088/mock_solver".to_string(),
+                "--price-estimation-drivers=test_solver|http://localhost:11088/test_solver"
+                    .to_string(),
+            ],
+        )
+        .await;
+    services
+        .start_api(vec![
+            "--price-estimation-drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        ])
+        .await;
+
+    // Place Orders
+    let order = OrderCreation {
+        sell_token: usdc.address(),
+        sell_amount: to_wei_with_exp(1000, 6),
+        buy_token: usdt.address(),
+        buy_amount: to_wei_with_exp(2000, 6),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    // Warm up co-located driver by quoting the order (otherwise placing an order
+    // may time out)
+    let _ = services
+        .submit_quote(&OrderQuoteRequest {
+            sell_token: usdc.address(),
+            buy_token: usdt.address(),
+            side: OrderQuoteSide::Sell {
+                sell_amount: SellAmount::BeforeFee {
+                    value: to_wei_with_exp(1000, 6).try_into().unwrap(),
+                },
+            },
+            ..Default::default()
+        })
+        .await;
+
+    let order_id = services.create_order(&order).await.unwrap();
+    let limit_order = services.get_order(&order_id).await.unwrap();
+    assert_eq!(limit_order.metadata.class, OrderClass::Limit);
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async {
+        // Keep mining blocks to trigger the event indexing logic
+        onchain.mint_block().await;
+        tokio::time::sleep(tokio::time::Duration::from_millis(1_000)).await;
+
+        let amm_usdc_balance_after = usdc.balance_of(USDC_WETH_COW_AMM).call().await.unwrap();
+        // CoW AMM traded automatically
+        amm_usdc_balance_after != amm_usdc_balance_before
+    })
+    .await
+    .unwrap();
+
+    // all cow amms on mainnet the helper contract is aware of
+    tracing::info!("Waiting for all cow amms to be indexed.");
+    let expected_cow_amms = [
+        addr!("027e1cbf2c299cba5eb8a2584910d04f1a8aa403"),
+        addr!("b3bf81714f704720dcb0351ff0d42eca61b069fc"),
+        addr!("301076c36e034948a747bb61bab9cd03f62672e3"),
+        addr!("d7cb8cc1b56356bb7b78d02e785ead28e2158660"),
+        addr!("9941fd7db2003308e7ee17b04400012278f12ac6"),
+        addr!("beef5afe88ef73337e5070ab2855d37dbf5493a4"),
+        addr!("c6b13d5e662fa0458f03995bcb824a1934aa895f"),
+    ];
+
+    wait_for_condition(TIMEOUT, || async {
```

Comment:
Since you are also adding the changes for fetching the native prices, could you also add the test I did for it (in which it proves it sends prices for tokens not present in the orders), or extend this one if possible?